ocaml/xapi/stream_vdi.ml

```diff
@@ -100,6 +100,64 @@ let write_block ~__context filename buffer ofd len =
     then raise (Api_errors.Server_error (Api_errors.client_error, [ExnHelper.string_of_exn e]))
     else raise e
 
+type descriptor = {
+  length : int64;
+  status_flags : int32;
+}
+
+let flag_hole = 1l
+let flag_zero = 2l
+
+let cycle_descriptors descriptor_list offset =
+  let rec range acc a b =
+    if a < b then acc
+    else range (a::acc) (Int64.add a Int64.minus_one) b
```

Comment:
You can simplify these: `Int64.(add a minus_one)`

Reply:
Good idea, using local module opens: http://dev.realworldocaml.org/files-modules-and-programs.html#scrollNav-6